### PR TITLE
Add a pre-commit hook

### PR DESCRIPTION
**Pull request by shaleh**

### CODE REVIEW

File: rust_src/admin/pre-commit [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+trap finish EXIT
+
+set -eu
+
+git stash save -q --keep-index $STASH_NAME
+
+check_char='\xE2\x9C\x93'
+cross_char='\xE2\x9D\x8C'
+green='\033[0;32m'
```
Comment (db48x) [inline rust_src/admin/pre-commit]: this could be $(tput setaf 2), so that it's not specific to any particular terminal.

Comment (shaleh) [reply rust_src/admin/pre-commit]: Or we could drop it. Either way.

**Comment by shaleh** [comment]
I dropped the colored output. It says success and FAIL! now. It should work everywhere. Support color blind people. The works :-)

**Comment by shaleh** [comment]
Note the OSX failure despite me touching NO code. YAY :-1:

**Comment by shaleh** [comment]
Toggling build.

**Pull request merged**